packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -406,9 +400,46 @@ - (void)updatePlayingState {
   _displayLink.running = _isPlaying || self.waitingForFrame;
 }
 
+- (void)sendFailedToLoadVideoEvent:(NSError *)error {
+  if (!NSThread.isMainThread) {
```

Comment:
it looks like `sendFailedToLoadVideoEvent` is only called from background queue. Can this function simply be: 
```
- (void)sendFailedToLoadVideoEventOnMainQueue:(NSError *)error {
  dispatch_async(dispatch_get_main_queue(), ^{
    // all logic here
  }
}
```

Reply:
It is also called from `setupEventSinkIfReadyToPlay` and `observeValueForKeyPath`.